python/cucumber_expressions/tree_regexp.py

```diff
@@ -51,22 +57,30 @@ def create_group_builder(self, regexp):
         return stack.pop()
 
     @staticmethod
-    def is_non_capturing(source, index):
-        # Regex is valid. Bounds check not required.
+    def is_named_group(source: str, index: int) -> bool:
+        """
+        Check if the group at the given index is a named capturing group, e.g. (?P<name>...).
+        """
+        return source[index + 1 : index + 3] == "P<" and source[index + 3] != "?"
+
+    @staticmethod
+    def extract_named_group_name(source: str, index: int) -> str:
+        """
+        Extract the name of a named capturing group, e.g., (?P<name>...) returns "name".
+        """
+        group_name_start = index + 3
+        group_name_end = source.find(">", group_name_start)
+        return source[group_name_start:group_name_end]
```

Comment:
My python is basically non existent, but above we're using `a : b` and here we're using `a:b` - Do they mean diff things if not maybe keep things standard?

Reply:
In the previous expression the spaces around `:` can be removed to keep codestyle consistent.
In python if you have a list, you can slice it with the `[start_pos:end_pos]` operator. It does not care if you have spaces around start_pos/end_pos.